networkx/classes/views.py

```diff
@@ -0,0 +1,275 @@
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, ddict in nbrs.items():
+                    d = ddict[data] if data in ddict else default
+                    yield (nbr, n, d)
+        else:  # data is False
+            for n, nbrs in self.nodes_nbrs:
+                for nbr in nbrs:
+                    yield (nbr, n)
+    
+
+class MultiEdgeView(EdgeView):
+    def __init__(self, graph, nbunch=None, keys=False,
+                 data=False, default=None):
+        super(MultiEdgeView, self).__init__(graph, nbunch, data, default)
+        self.keys = keys
+    def __iter__(self):
+        seen = {}
+        data = self.data
+        keys = self.keys
+        if data is True:
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    if nbr not in seen:
+                        for key, ddict in keydict.items():
+                            if keys:
+                                yield (n, nbr, key, ddict)
+                            else:
+                                yield (n, nbr, ddict)
+                seen[n] = 1
+        elif data is not False:
+            default = self.default
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    if nbr not in seen:
+                        for key, ddict in keydict.items():
+                            d = ddict[data] if data in ddict else default
+                            yield (n, nbr, key, d) if keys else (n, nbr, d)
+                seen[n] = 1
+        else:
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    if nbr not in seen:
+                        for key in keydict:
+                            yield (n, nbr, key) if keys else (n, nbr)
+                seen[n] = 1
+        del seen
+
+class MultiDiEdgeView(MultiEdgeView):
+    def __iter__(self):
+        data = self.data
+        keys = self.keys
+        if data is True:
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    for key, ddict in keydict.items():
+                        yield (n, nbr, key, ddict) if keys else (n, nbr, ddict)
+        elif data is not False:
+            default = self.default
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    for key, ddict in keydict.items():
+                        d = ddict[data] if data in ddict else default
+                        yield (n, nbr, key, d) if keys else (n, nbr, d)
+        else:
+            for n, nbrs in self.nodes_nbrs:
+                for nbr, keydict in nbrs.items():
+                    for key in keydict:
+                        yield (n, nbr, key) if keys else (n, nbr)
+
+class InMultiDiEdgeView(InDiEdgeView):
+    def __init__(self, graph, nbunch=None, keys=False, data=False, default=None):
```

Comment:
The signature doesn't match the call in MultiDiGraph views, should be data first then keys.
``` python
def __init__(self, graph, nbunch=None, data=False, keys=False, default=None):
```

[The reason for travis failures on MultiDiGraphs]